dev/small-requirements.txt

```diff
@@ -2,25 +2,25 @@
 # Required
 azure-storage-blob==12.3.0
 # Required to log artifacts and models to GCS artifact locations
-google-cloud-storage==1.14.0
+google-cloud-storage
 # Required to log artifacts and models to AWS S3 artifact locations
-botocore==1.12.84  # pinned for moto
-boto3==1.9.84      # pinned for moto
+botocore
+boto3
 # Required to log artifacts and models to HDFS artifact locations
-pyarrow==0.17.0
+pyarrow
 # Required to log artifacts to SFTP artifact locations
-pysftp==0.2.9
+pysftp
 # Required to run the MLflow server against SQL-backed storage
-sqlalchemy==1.3.0
+sqlalchemy
 # Required by the mlflow.projects module, when running projects against
 # a remote Kubernetes cluster
-kubernetes==9.0.0
+kubernetes
 ## Test-only dependencies
-attrdict==2.0.0
+attrdict
```

Comment:
I would rather we remove all of these from `small-requirements` and instead add the comments about specific packages to `setup.py`.